### PR TITLE
Enable wolfSSL SP Math all (sp_int.c) by default

### PR DESCRIPTION
**Pull request by dgarske**
* SP Math (default) (sp_int.c): `./configure` or `./configure --enable-sp-math-all`
* Fast Math (tfm.c): `./configure --enable-fastmath`
* Heap Math (integer.c): `./configure --enable-heapmath`

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -452,7 +443,7 @@ AC_ARG_ENABLE([sp],
 AC_ARG_ENABLE([sp-math-all],
     [AS_HELP_STRING([--enable-sp-math-all],[Enable Single Precision math implementation for full algorithm suite (default: disabled)])],
```
Comment (SparkiDev) [inline configure.ac:444]: Changed the default to 'enabled'.

File: configure.ac [hunk 2]
```diff
@@ -5496,6 +5487,17 @@ then
     fi
 fi
 
+# old integer.c math
+AC_ARG_ENABLE([oldmath],
```
Comment (SparkiDev) [inline configure.ac:5491]: I would prefer 'heapmath' to 'oldmath' - more descriptive.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Comment by SparkiDev** [comment]
Looks like I used sp_digit instead of sp_int_digit in sp_int.c.

**Comment by SparkiDev** [comment]
Casting required for Windows as well.

**Comment by dgarske** [comment]
@SparkiDev I think this is in a good state now. If there are any issues please go ahead and fix them and push to this PR. Todd asked you finalize the SP math all by default work. Thanks!

**Comment by anhu** [comment]
So, from some of the comments I'm getting the impression that fips customers should not be using SP-math.  Is that impression correct?

**Comment by kaleb-himes** [comment]
@SparkiDev happy to help on this if you needed, I've dealt a lot with SP math and FIPS, let's sync up when you start work for the day. I'll start testing things in the meantime and see if I can't narrow it down, I believe this may be a little tricky, we may need to hinge the sp math by default on a few conditions:

is it `HAVE_FIPS_VERSION == 2` and if so is the module version one of the modules that supports sp math (versions listed above in my reply to @anhu )

**Review by SparkiDev — APPROVED** [review]

**Comment by dgarske** [comment]
The state of this PR looks awesome. Would be nice to finalize this. Seems to just be a a Windows FIPS ACVP issue with build settings.

@kaleb-himes can you review the errors in this failed test case?
https://test.wolfssl.com/jenkins/job/PRB-FIPS-windows-test-part2-ACVP-v2/1076/

**Comment by dgarske** [comment]
Kaleb provided me a FIPS patch and I fixed the build issue. Let's see what Jenkins thinks now. Assigning back to me for finalization.

**Review by kaleb-himes — APPROVED** [review]

**Review by dgarske — COMMENTED** [review]
Approved. I opened PR, so can't mark approved, but can merge it. Get ready everyone!

**Pull request merged**